src/python/pants/help/help_info_extracter.py

```diff
@@ -79,7 +80,8 @@ class OptionHelpInfo:
                             scope context (e.g., [--baz, --no-baz, --qux])
     env_var: The environment variable that set's the option.
     config_key: The config key for this option (in the section named for its scope).
-
+    target_field_name: The name for the field that overrides this option in `local_environment`,
```

Comment:
This name should maybe indicate that the field is "environment aware"? It's obvious from the help string, but...

Reply:
To capture both pieces of feedback, I'll add an `environment_aware` flag to the dataclass here.

Reply:
It looks like `generate_docs` would need a bit of work to be able to facilitate a link to the environments topic. Going to leave that as a separate ticket.

Reply:
See #17126